### ISSUE TITLE
prefer-const incorrectly reports assignments in a nested lexical scope

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1214]
`prefer-const` reports a variable whose declaration and only assignment are in different lexical scopes. Changing the declaration to `const` is not valid, and moving it into the nested block would make the later read inaccessible.

Reproduced with `@utoo/lint@0.4.0`, Node.js 22.23.1, macOS arm64. ESLint 8.57.1 reports no diagnostics for the same input and rule configuration.

### Reproduction

Install `@utoo/lint@0.4.0` and save this as `repro.cjs`:

```js
const { ESLint } = require('@utoo/lint');

const source = `let value;
try {
  value = 1;
} catch (error) {
  throw error;
}
console.log(value);
`;

(async () => {
  const engine = new ESLint({
    useEslintrc: false,
    overrideConfig: {
      parserOptions: { ecmaVersion: 2022 },
      rules: { 'prefer-const': 'error' }
    }
  });
  const [result] = await engine.lintText(source, {
    filePath: 'input.js'
  });
  console.log(result.messages);
})();
```

### Actual

An error is reported at line 3, column 3:

```text
'value' is never reassigned. Use 'const' instead.
```

### Expected

No `prefer-const` diagnostic, matching ESLint. The assignment is in a nested scope and cannot be combined with the outer declaration as a const initialization while preserving scope.

Also checked with `fix: true`: there is no generated fix, but the erroneous diagnostic remains and can fail CI.
